File: server/Services/UserRepository.cs
using System.Threading.Tasks;
using plantio.Domain;

namespace plantio.Services {
    public interface UserRepository {
        Task<bool> ExistsWithName(string name);
        Task<bool> Save(User user);
        Task<bool> Update(User user);
        void Delete(User user);
        Task<User?> GetByName(string name);
    }
}

File: plantio/src/main/java/com/deltegui/plantio/users/application/UserRepository.java
package com.deltegui.plantio.users.application;

import com.deltegui.plantio.users.domain.User;

import java.util.List;
import java.util.Optional;

public interface UserRepository {
    Optional<User> findByName(String name);
    void delete(User user);
    void save(User user);
    void update(User user);
    List<User> getAll();
}
